tools/cmd.py

```diff
@@ -0,0 +1,510 @@
+# Copyright (c) 2017-2019 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from datetime import datetime
+from enum import IntEnum
+import os
+import re
+import socket
+import struct
+from tools.scp import SCP
+
+
+def _word(a, b, c, d):
+    """
+    Assembles a word out of four bytes.
+
+    aaaaaaaa------------------------   << 24
+    --------bbbbbbbb----------------   << 16
+    ----------------cccccccc--------   <<  8
+    ------------------------dddddddd   <<  0
+    """
+    return (
+        (a & 0xFF) << 24) | ((b & 0xFF) << 16) | ((c & 0xFF) << 8) | (d & 0xFF)
+
+
+# -----------------------------------------------------------------------------
+
+
+class CMD(IntEnum):
+    VER = 0
+    RUN = 1
+    READ = 2
+    WRITE = 3
+    FILL = 5
+
+    LINK_READ = 17
+    LINK_WRITE = 18
+
+    LED = 25
+    IPTAG = 26
+
+    # Tubotron message
+    TUBE = 64
+
+
+class SCAMP_CMD(CMD):
```

Comment:
Computer says NO!
TypeError: Cannot extend enumerations
https://docs.python.org/3/library/enum.html#restricted-subclassing-of-enumerations

Reply:
Fixed